From 80948898b3ff04b52c9bbe1394abf1f1fd17e1f5 Mon Sep 17 00:00:00 2001
From: Brenden Tisler <btisler@emacinc.com>
Date: Tue, 25 Jul 2023 14:54:59 -0500
Subject: [PATCH] Fix ethernet to use the correct phy-mode

It was found in the kernel that the phy mode needed to be set to rgmii-rxid due to the cpsw switch
This fixed it here too
Also changed default console env var to the correct standard debug port

Signed-off-by: Brenden Tisler <btisler@emacinc.com>
---
 arch/arm/dts/emac-som-5728m.dtsi | 25 +++++++++++++++----------
 board/emacinc/som-5728m/board.c  | 14 +++++++++-----
 configs/emac-som-5728m_defconfig |  1 -
 3 files changed, 24 insertions(+), 16 deletions(-)

diff --git a/arch/arm/dts/emac-som-5728m.dtsi b/arch/arm/dts/emac-som-5728m.dtsi
index d074b368d6..eb9235d554 100644
--- a/arch/arm/dts/emac-som-5728m.dtsi
+++ b/arch/arm/dts/emac-som-5728m.dtsi
@@ -450,14 +450,19 @@
 };
 
 &davinci_mdio {
-	reset-gpios = <&gpio4 25 GPIO_ACTIVE_LOW>;
-	reset-delay-us = <2>;
-
-	phy0: ethernet-phy@1 {
-		reg = <4>;
-		eee-broken-100tx;
-		eee-broken-1000t;
-	};
+	ethphy4: ethernet-phy@4 {
+        reg = <4>;
+		reset-gpios = <&gpio4 25 GPIO_ACTIVE_LOW>;
+		reset-assert-us = <50000>;
+		reset-deassert-us = <50000>;
+		qca,clk-out-frequency = <50000000>;
+		qca,keep-pll-enabled;
+		vddio-supply = <&vddio>;
+		vddio: vddio-regulator {
+			regulator-min-microvolt = <1500000>;
+			regulator-max-microvolt = <1500000>;
+		};
+    };
 };
 
 &mac {
@@ -466,8 +471,8 @@
 };
 
 &cpsw_emac0 {
-	phy-handle = <&phy0>;
-	phy-mode = "rgmii";
+	phy-handle = <&ethphy4>;
+	phy-mode = "rgmii-rxid";
 };
 
 &usb2_phy1 {
diff --git a/board/emacinc/som-5728m/board.c b/board/emacinc/som-5728m/board.c
index bfd35059ef..c51a2a2575 100644
--- a/board/emacinc/som-5728m/board.c
+++ b/board/emacinc/som-5728m/board.c
@@ -388,7 +388,7 @@ int board_late_init(void)
 	uclass_get_device(UCLASS_CLK, 0, &dev);
 
 	if (board_is_som_5728m())
-		env_set("console", "ttyS0,115200n8");
+		env_set("console", "ttyS2,115200n8");
 
 #if CONFIG_IS_ENABLED(DM_USB) && CONFIG_IS_ENABLED(OF_CONTROL)
 	if (device_okay("/ocp/omap_dwc3_1@48880000"))
@@ -399,13 +399,17 @@ int board_late_init(void)
 	return 0;
 }
 
-int last_stage_init(void) {
-	udelay(100000);
+int board_phy_config(struct phy_device *phydev){
+	udelay(50000);
 	gpio_request(GPIO_PHY_RST, "phy_rst");
 	gpio_direction_output(GPIO_PHY_RST, 0);
-	udelay(500000);
+	udelay(50000);
 	gpio_direction_output(GPIO_PHY_RST, 1);
-	udelay(3000000);
+	udelay(50000);
+
+	if (phydev->drv->config)
+		phydev->drv->config(phydev);
+	
 	return 0;
 }
 
diff --git a/configs/emac-som-5728m_defconfig b/configs/emac-som-5728m_defconfig
index 8608134172..cefa8c5166 100644
--- a/configs/emac-som-5728m_defconfig
+++ b/configs/emac-som-5728m_defconfig
@@ -29,7 +29,6 @@ CONFIG_BOOTARGS_SUBST=y
 # CONFIG_USE_BOOTCOMMAND is not set
 CONFIG_SYS_CONSOLE_INFO_QUIET=y
 CONFIG_BOARD_EARLY_INIT_F=y
-CONFIG_LAST_STAGE_INIT=y
 # CONFIG_MISC_INIT_R is not set
 CONFIG_SPL_MAX_SIZE=0x7bc00
 CONFIG_SPL_SYS_MALLOC_SIMPLE=y
